lib/internal/test_runner/utils.js

```diff
@@ -150,17 +148,17 @@ async function getReportersMap(reporters, destinations) {
 }
 
 
-async function setupTestReporters(testsStream) {
-  try {
-    const { reporters, destinations } = parseCommandLine();
-    const reportersMap = await getReportersMap(reporters, destinations);
-    for (let i = 0; i < reportersMap.length; i++) {
-      const { reporter, destination } = reportersMap[i];
-      compose(testsStream, reporter).pipe(destination);
-    }
-  } catch (err) {
-    throw new ERR_TEST_FAILURE(err, kAsyncBootstrapFailure);
+async function setupTestReporters(rootTest) {
+  const { reporters, destinations } = parseCommandLine();
+  const reportersMap = await getReportersMap(reporters, destinations);
+  for (let i = 0; i < reportersMap.length; i++) {
+    const { reporter, destination } = reportersMap[i];
+    compose(rootTest.reporter, reporter).pipe(destination);
   }
+
+  // Setting up reporters is currently the last step of the bootstrapping
+  // process. If that ever changes, this code will need to move.
+  rootTest.harness.bootstrapComplete = true;
```

Comment:
this assumes `setupTestReporters` will always be called, but when running using `run` it won't be called (then all uncaught errors will be rethrown)